src/earn/EarnEnterAmount.tsx

```diff
@@ -202,30 +231,42 @@ function EarnEnterAmount({ route }: Props) {
         localAmount: parsedLocalAmount,
       }
     }
-  }, [tokenAmountInput, localAmountInput, enteredIn, token])
+  }, [tokenAmountInput, localAmountInput, enteredIn, inputToken, mode, withdrawToken])
 
-  const feeCurrencies = useSelector((state) => feeCurrenciesSelector(state, token.networkId))
+  const feeCurrencies = useSelector((state) =>
+    feeCurrenciesSelector(
+      state,
+      mode === 'withdraw' ? withdrawToken.networkId : inputToken.networkId
+    )
+  )
 
   useEffect(() => {
     clearPreparedTransactions()
 
     if (
       !tokenAmount ||
       tokenAmount.isLessThanOrEqualTo(0) ||
-      tokenAmount.isGreaterThan(token.balance)
+      tokenAmount.isGreaterThan(isWithdrawal ? withdrawToken.balance : inputToken.balance)
     ) {
       return
     }
     const debouncedRefreshTransactions = setTimeout(() => {
-      return handleRefreshPreparedTransactions(tokenAmount, token, feeCurrencies)
+      return handleRefreshPreparedTransactions(
+        tokenAmount,
+        isWithdrawal ? withdrawToken : inputToken,
```

Comment:
all of these variations of `isWithdrawal ? withdrawToken : inputToken` throughout the component make the code a bit difficult to follow. Can you make a variable for it? Perhaps `transactionToken` is correct since its the token that will ultimately be used in the transaction?

Reply:
Adjusted in 41965058d4ae86e34e123cf5528867986cede2d0!